src/main/java/com/liveramp/hyperminhash/HmhCardinalityEstimator.java

```diff
@@ -965,15 +967,13 @@ private static boolean useLinearCounting(final int p, final long linearCountEsti
     }
   }
 
-  private static long basicHllEstimate(long[] packedRegisters, int r) {
+  private static long basicHllEstimate(long[] registers, int p, int r) {
     double denominator = 0.0;
-    for (long register : packedRegisters) {
+    for (long register : registers) {
       denominator += Math.pow(2, -1 * LongPacker.unpackPositionOfFirstOne(register, r));
     }
 
-    final double numerator = alpha(packedRegisters.length)
```

Comment:
this was a bug where we were passing M instead of P and was causing us to severely undercount